symfony/plugins/orangehrmPimPlugin/Api/EmpUsTaxExemptionAPI.php

```diff
@@ -0,0 +1,241 @@
+        $workStateCode = $this->getRequestParams()->getStringOrNull(RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_WORK_STATE_CODE);
+        $empUsTaxExemption = $this->getEmpUsTaxExemptionService()->getEmpUsTaxExemptionDao()->getEmployeeTaxExemption($empNumber);
+        if (!$empUsTaxExemption instanceof EmpUsTaxExemption) {
+            $empUsTaxExemption = new EmpUsTaxExemption();
+            $empUsTaxExemption->getDecorator()->setEmployeeByEmpNumber($empNumber);
+        }
+        $empUsTaxExemption->setFederalStatus($federalStatus);
+        $empUsTaxExemption->setFederalExemptions($federalExemptions);
+        $empUsTaxExemption->setState($taxStateCode);
+        $empUsTaxExemption->setStateStatus($stateStatus);
+        $empUsTaxExemption->setStateExemptions($stateExemptions);
+        $empUsTaxExemption->setUnemploymentState($unemploymentStateCode);
+        $empUsTaxExemption->setWorkState($workStateCode);
+        return $this->getEmpUsTaxExemptionService()->getEmpUsTaxExemptionDao()->saveEmployeeTaxExemption($empUsTaxExemption);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForUpdate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_EMP_NUMBER,
+                new Rule(Rules::IN_ACCESSIBLE_EMP_NUMBERS)
+            ),
+            new ParamRule(
+                CommonParams::PARAMETER_ID,
+                new Rule(Rules::BETWEEN, [0, 100])
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(self::PARAMETER_FEDERAL_STATUS,
+                    new Rule(Rules::STRING_TYPE),
+                    new Rule(Rules::IN, [
+                        [
+                            EmpUsTaxExemption::FILTER_SINGLE,
+                            EmpUsTaxExemption::FILTER_MARRIED,
+                            EmpUsTaxExemption::FILTER_NON_RESIDENT_ALIEN,
+                            EmpUsTaxExemption::FILTER_NOT_APPLICABLE
+                        ]
+                    ]),
+                ),
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(self::PARAMETER_FEDERAL_EXEMPTIONS,
+                    new Rule(Rules::BETWEEN, [0, 100]),
+                ),
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(self::PARAMETER_TAX_STATE_CODE,
+                    new Rule(Rules::STRING_TYPE),
+                    new Rule(Rules::PROVINCE_CODE),
+                ),
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(self::PARAMETER_STATE_STATUS,
+                    new Rule(Rules::STRING_TYPE),
+                    new Rule(Rules::IN, [
+                        [
+                            EmpUsTaxExemption::FILTER_SINGLE,
```

Comment:
Shall we move these into `EmpUsTaxExemption::STATUSES`
And will refactor `FILTER_` to `STATUS_` in each const